lightning-invoice/src/de.rs

```diff
@@ -553,6 +556,11 @@ impl FromBase32 for Fallback {
 					return Err(ParseError::InvalidSegWitProgramLength);
 				}
 
+				let version = match WitnessVersion::try_from(version) {
+					Ok(version) => version,
+					Err(_) => return Err(ParseError::InvalidSegWitVersion),
```

Comment:
Could you add deserialization test case below for this new error?

Reply:
good suggestion it turns out this error can't happen so I got rid of it!